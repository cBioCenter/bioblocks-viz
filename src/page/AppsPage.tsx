import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { Button, Grid, Header, List } from 'semantic-ui-react';

import { IVizSummaryData, VizData } from '~chell-viz~/data';

export interface IAppsPageProps extends Partial<RouteComponentProps> {}

export class AppsPage extends React.Component<IAppsPageProps, any> {
  constructor(props: IAppsPageProps) {
    super(props);
  }

  public render() {
    return (
      <List divided={true}>
        <List.Item>{this.renderAppItem(VizData.spring)}</List.Item>
        <List.Item>{this.renderAppItem(VizData.tfjsTsne)}</List.Item>
        <List.Item>{this.renderAppItem(VizData.anatomogram)}</List.Item>
      </List>
    );
  }

  protected renderAppItem(viz: IVizSummaryData) {
    return (
      <Grid centered={true} columns={3} padded={true} relaxed={true}>
        <Grid.Column>
          <img
            src={`assets/icons/${viz.name.toLocaleLowerCase()}-thumbnail.png`}
            alt={`icon for ${viz.name}`}
            style={{ height: '100px' }}
          />
        </Grid.Column>
        <Grid.Column textAlign={'left'}>
          <Header as={'h2'}>{viz.name}</Header>
          <List>
            <List.Item>{`${viz.listAsOriginal ? 'original: ' : ''}${viz.authors.join(', ')}`}</List.Item>
            <List.Item>{viz.summary}</List.Item>
            <List.Item>{`relevant data: ${viz.relevantData}`}</List.Item>
          </List>
        </Grid.Column>
        <Grid.Column stretched={true}>
          <Grid.Row>
            <Button basic={true}>
              <Link to={{ pathname: '/overview', search: `?name=${viz.name.toLocaleLowerCase()}` }}>details</Link>
            </Button>
          </Grid.Row>
          <Grid.Row>
            <Button basic={true}>
              <Link to={'/beta'}>launch</Link>
            </Button>
          </Grid.Row>
        </Grid.Column>
      </Grid>
    );
  }
}
