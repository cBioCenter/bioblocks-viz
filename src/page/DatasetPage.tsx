import { fromJS, List } from 'immutable';
import * as React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { bindActionCreators, Dispatch } from 'redux';
import { Grid, Message } from 'semantic-ui-react';

import { fetchLabeledSpringData, LabeledCellsActions } from '~chell-viz~/action';
import { AnatomogramContainer, SpringContainer, TensorTContainer } from '~chell-viz~/container';
import { SPECIES_TYPE, VizData } from '~chell-viz~/data';

export interface IDatasetPageProps extends Partial<RouteComponentProps> {
  fetchSpringData(datasetLocation: string): void;
  setSpecies(species: SPECIES_TYPE): void;
}

export interface IDatasetPageState {
  visualizations: List<string>;
  datasetLocation: string;
}

class DatasetPageClass extends React.Component<IDatasetPageProps, IDatasetPageState> {
  constructor(props: IDatasetPageProps) {
    super(props);
    this.state = {
      datasetLocation: '',
      visualizations: List<string>(),
    };
  }

  public componentDidMount() {
    if (this.props.location) {
      this.setupSearchParameters(this.props.location.search);
    }
  }

  public componentDidUpdate(prevProps: IDatasetPageProps, prevState: IDatasetPageState) {
    const { location, setSpecies } = this.props;
    const { datasetLocation } = this.state;
    if (location && location !== prevProps.location) {
      this.setupSearchParameters(location.search);
    } else if (datasetLocation !== prevState.datasetLocation) {
      setSpecies(datasetLocation.includes('hpc') ? 'homo_sapiens' : 'mus_musculus');
    }
  }

  public render() {
    const { visualizations, datasetLocation } = this.state;

    return (
      <div style={{ padding: '20px' }}>
        <Grid centered={true} stackable={true} stretched={false} padded={true} columns={2}>
          {datasetLocation.length >= 1 &&
            visualizations.map((visualization, index) => (
              <Grid.Column key={`dataset-visualization-${index}`} style={{ width: 'auto' }}>
                {this.renderVisualization(visualization, datasetLocation)}
              </Grid.Column>
            ))}
        </Grid>
      </div>
    );
  }

  protected setupSearchParameters(query: string) {
    const { fetchSpringData } = this.props;
    const params = new URLSearchParams(query);
    // tslint:disable-next-line:no-backbone-get-set-outside-model
    const datasetLocation = params.get('name');
    const visualizations = fromJS(params.getAll('viz')) as List<string>;

    fetchSpringData(datasetLocation ? datasetLocation : '');
    this.setState({
      datasetLocation: datasetLocation ? datasetLocation : '',
      visualizations,
    });
  }

  protected renderVisualization(viz: string | undefined, datasetLocation: string) {
    const isFullPage = this.state.visualizations.size === 1;
    switch (viz) {
      case VizData.spring.name.toLocaleLowerCase():
        return <SpringContainer datasetLocation={datasetLocation} isFullPage={isFullPage} />;
      case VizData.tfjsTsne.name.toLocaleLowerCase():
        return <TensorTContainer datasetLocation={datasetLocation} isFullPage={isFullPage} />;
      case VizData.anatomogram.name.toLocaleLowerCase():
        return <AnatomogramContainer />;
      default:
        return <Message error={true}>{`Currently unsupported visualization '${viz}'`}</Message>;
    }
  }
}

const UnconnectedDatasetPage = (props: IDatasetPageProps) => <DatasetPageClass {...props} />;

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      fetchSpringData: fetchLabeledSpringData,
      setSpecies: LabeledCellsActions.setSpecies,
    },
    dispatch,
  );

// tslint:disable-next-line:max-classes-per-file
export class DatasetPage extends connect(
  undefined,
  mapDispatchToProps,
)(UnconnectedDatasetPage) {}
