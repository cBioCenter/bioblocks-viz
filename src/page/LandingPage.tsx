import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { Button, Container, Divider, Grid, Header, Segment } from 'semantic-ui-react';

import { IDatasetInfo, IStory, IVizSummaryData, Stories, userDatasets, VizData } from '~chell-viz~/data';

export interface ILandingPageProps extends Partial<RouteComponentProps> {}

export class LandingPage extends React.Component<ILandingPageProps, any> {
  constructor(props: ILandingPageProps) {
    super(props);
  }

  public render() {
    return (
      <Container fluid={false}>
        <Segment basic={true} padded={'very'}>
          <Grid centered={true} padded={true} relaxed={true}>
            {this.renderFeaturedStories()}
            {this.renderFeaturedVisualizations()}
            {this.renderFeaturedDatasets()}
          </Grid>
        </Segment>
      </Container>
    );
  }

  protected renderSingleVisualization(viz: IVizSummaryData) {
    return (
      <Grid.Row columns={3}>
        <Grid.Column width={2}>
          <img
            src={`assets/icons/${viz.name.toLocaleLowerCase()}-thumbnail.png`}
            style={{ height: '90px', width: '90px' }}
            alt={`viz ${viz.name} icon`}
          />
        </Grid.Column>
        <Grid.Column textAlign={'left'} width={8}>
          <Header>{viz.name}</Header>
          {viz.summary}
          <p>
            <span style={{ fontWeight: 'bold' }}>{viz.listAsOriginal ? 'Original authors' : 'Authors'}: </span>
            {viz.authors.length === 2 ? `${viz.authors[0]} and ${viz.authors[1]}` : viz.authors.join(', ')}
          </p>
        </Grid.Column>
        <Grid.Column width={2}>
          <Grid.Row>
            <Button basic={true}>
              <Link to={{ pathname: '/visualizations/', search: `?name=${viz.name.toLocaleLowerCase()}` }}>
                details
              </Link>
            </Button>
          </Grid.Row>
          <Grid.Row>
            <Button basic={true}>
              <Link to={{ pathname: '/dataset', search: `?viz=${viz.name.toLocaleLowerCase()}` }}>launch</Link>
            </Button>
          </Grid.Row>
        </Grid.Column>
      </Grid.Row>
    );
  }

  protected renderFeaturedDatasets() {
    return (
      <>
        <Grid.Row centered={true}>
          <Grid.Column width={12}>
            <Header floated={'left'}>Featured Datasets</Header>
            <Divider section={true} />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row columns={2} divided={true}>
          {this.renderHCADatasets()}
          {this.renderUserSharedDatasets()}
        </Grid.Row>
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Link style={{ color: 'blue', float: 'right' }} to={'datasets'}>
              more datasets...
            </Link>
          </Grid.Column>
        </Grid.Row>
      </>
    );
  }

  protected renderFeaturedStories() {
    return (
      <>
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Header floated={'left'}>Featured Stories</Header>
            <Divider section={true} />
          </Grid.Column>
        </Grid.Row>
        {this.renderSingleFeaturedStory(Stories[0])}
        {this.renderSingleFeaturedStory(Stories[1])}
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Link style={{ color: 'blue', float: 'right' }} to={'stories'}>
              more stories...
            </Link>
          </Grid.Column>
        </Grid.Row>
      </>
    );
  }

  protected renderFeaturedVisualizations() {
    return (
      <>
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Header floated={'left'}>Featured Visualizations</Header>
            <Divider section={true} />
          </Grid.Column>
        </Grid.Row>
        {this.renderSingleVisualization(VizData.spring)}
        {this.renderSingleVisualization(VizData.tfjsTsne)}
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Link style={{ color: 'blue', float: 'right' }} to={'visualizations'}>
              more visualizations...
            </Link>
          </Grid.Column>
        </Grid.Row>
      </>
    );
  }

  protected renderSingleFeaturedStory(story: IStory) {
    return (
      <Grid.Row columns={3}>
        <Grid.Column width={2}>
          <img src={story.icon} style={{ height: '90px', width: '90px' }} alt={`story ${story.title} icon`} />
        </Grid.Column>
        <Grid.Column textAlign={'left'} width={8}>
          <Header>{story.title}</Header>
          <p>
            <span style={{ fontWeight: 'bold' }}>Description: </span>
            {story.description}
          </p>
          <p>
            <span style={{ fontWeight: 'bold' }}>Analysis authors: </span>
            {story.authors.length === 2 ? `${story.authors[0]} and ${story.authors[1]}` : story.authors.join(', ')}
          </p>
        </Grid.Column>
        <Grid.Column width={2}>
          <Button basic={true}>
            <Link to={story.link}>launch</Link>
          </Button>
        </Grid.Column>
      </Grid.Row>
    );
  }

  protected renderHCADatasets() {
    return (
      <Grid.Column width={3}>
        <Grid.Row>
          <img
            src={'assets/icons/hca-logo.png'}
            style={{ width: '150px', height: '250px' }}
            alt={'human cell atlas logo'}
          />
          <Grid.Row>
            <a href={'https://preview.data.humancellatlas.org/'}>Search HCA datasets ...</a>
          </Grid.Row>
        </Grid.Row>
      </Grid.Column>
    );
  }

  protected renderSingleUserSharedDatasets(dataset: IDatasetInfo) {
    return (
      <Grid.Row columns={2}>
        <Grid.Column floated={'left'}>
          <Header>{dataset.name}</Header>
          <p>
            {dataset.summary}
            <br />
            <span style={{ fontWeight: 'bold' }}>Authors: </span>
            {dataset.authors.join(', ')}
          </p>
        </Grid.Column>
        <Grid.Column floated={'right'}>
          <Button basic={true} floated={'right'}>
            <Link to={dataset.links.detail}>details</Link>
          </Button>
          <Button basic={true} floated={'right'}>
            <Link to={dataset.links.analysis}>analyze</Link>
          </Button>
        </Grid.Column>
      </Grid.Row>
    );
  }

  protected renderUserSharedDatasets() {
    return (
      <Grid.Column stretched={true} textAlign={'left'} width={8}>
        <Grid.Row textAlign={'center'}>
          <Header as={'h2'} textAlign={'center'}>
            User shared
          </Header>
        </Grid.Row>
        <br />
        <Grid.Row>{this.renderSingleUserSharedDatasets(userDatasets[0])}</Grid.Row>
        <Grid.Row>{this.renderSingleUserSharedDatasets(userDatasets[1])}</Grid.Row>
      </Grid.Column>
    );
  }
}
