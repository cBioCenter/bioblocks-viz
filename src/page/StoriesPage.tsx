import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { Button, Grid, Header } from 'semantic-ui-react';

import { IStory, Stories } from '~chell-viz~/data';

export interface IStoriesPageProps extends Partial<RouteComponentProps> {}

export class StoriesPage extends React.Component<IStoriesPageProps, any> {
  constructor(props: IStoriesPageProps) {
    super(props);
  }

  public render() {
    return (
      <Grid centered={true} divided={'vertically'} padded={true} relaxed={false}>
        {Stories.map((story, index) => (
          <React.Fragment key={`story-${index}`}>{this.renderSingleFeaturedStory(story)}</React.Fragment>
        ))}
      </Grid>
    );
  }

  protected renderSingleFeaturedStory(story: IStory) {
    return (
      <Grid.Row columns={3}>
        <Grid.Column>
          <img src={story.icon} style={{ height: '90px', width: '90px' }} alt={`story ${story.title} icon`} />
        </Grid.Column>
        <Grid.Column textAlign={'left'}>
          <Header>{story.title}</Header>
          <p>
            <span style={{ fontWeight: 'bold' }}>Description: </span>
            {story.description}
          </p>
          <p>
            <span style={{ fontWeight: 'bold' }}>Analysis authors: </span>
            {story.authors.length === 2 ? `${story.authors[0]} and ${story.authors[1]}` : story.authors.join(', ')}
          </p>
          <br />
        </Grid.Column>
        <Grid.Column>
          <Button basic={true}>
            <Link to={story.link}>launch</Link>
          </Button>
        </Grid.Column>
      </Grid.Row>
    );
  }
}
