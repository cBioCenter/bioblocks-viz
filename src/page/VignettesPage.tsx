import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { Button, Grid, Header } from 'semantic-ui-react';

import { IVignette, Vignettes } from '~chell-viz~/data';

export interface IVignettesPageProps extends Partial<RouteComponentProps> {}

export class VignettesPage extends React.Component<IVignettesPageProps, any> {
  constructor(props: IVignettesPageProps) {
    super(props);
  }

  public render() {
    return (
      <Grid centered={true} padded={true} relaxed={false}>
        {Vignettes.map((vignette, index) => (
          <React.Fragment key={`vignette-${index}`}>{this.renderSingleFeaturedVignette(vignette)}</React.Fragment>
        ))}
      </Grid>
    );
  }

  protected renderSingleFeaturedVignette(vignette: IVignette) {
    return (
      <Grid.Row columns={3}>
        <Grid.Column>
          <img src={vignette.icon} style={{ height: '90px', width: '90px' }} alt={`vignette ${vignette.title} icon`} />
        </Grid.Column>
        <Grid.Column textAlign={'left'}>
          <Header>{vignette.title}</Header>
          <p>
            <span style={{ fontWeight: 'bold' }}>Description: </span>
            {vignette.description}
          </p>
          <p>
            <span style={{ fontWeight: 'bold' }}>Analysis authors: </span>
            {vignette.authors.length === 2
              ? `${vignette.authors[0]} and ${vignette.authors[1]}`
              : vignette.authors.join(', ')}
          </p>
        </Grid.Column>
        <Grid.Column>
          <Button basic={true}>
            <Link to={vignette.link}>launch</Link>
          </Button>
        </Grid.Column>
      </Grid.Row>
    );
  }
}
