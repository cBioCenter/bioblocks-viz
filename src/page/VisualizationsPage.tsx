import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { Button, Grid, Header, List } from 'semantic-ui-react';

import { IVizSummaryData, VizData } from '~bioblocks-viz~/data';

export interface IVisualizationsPageProps extends Partial<RouteComponentProps> {}

export class VisualizationsPage extends React.Component<IVisualizationsPageProps, any> {
  constructor(props: IVisualizationsPageProps) {
    super(props);
  }

  public render() {
    return (
      <List divided={true}>
        <List.Item>{this.renderVisualizationItem(VizData.spring)}</List.Item>
        <List.Item>{this.renderVisualizationItem(VizData.tfjsTsne)}</List.Item>
        <List.Item>{this.renderVisualizationItem(VizData.anatomogram)}</List.Item>
      </List>
    );
  }

  protected renderVisualizationItem(viz: IVizSummaryData) {
    return (
      <Grid centered={true} columns={3} padded={true} relaxed={true}>
        <Grid.Column>
          <img
            src={`assets/icons/${viz.name.toLocaleLowerCase()}-thumbnail.png`}
            alt={`icon for ${viz.name}`}
            style={{ height: '100px' }}
          />
        </Grid.Column>
        <Grid.Column textAlign={'left'}>
          <Header as={'h2'}>{viz.name}</Header>
          <List>
            <List.Item>{`${viz.listAsOriginal ? 'original: ' : ''}${viz.authors.join(', ')}`}</List.Item>
            <List.Item>{viz.summary}</List.Item>
            <List.Item>{`relevant data: ${viz.relevantData}`}</List.Item>
          </List>
        </Grid.Column>
        <Grid.Column stretched={true}>
          <Grid.Row>
            <Button basic={true}>
              <Link to={{ pathname: '/visualizations/', search: `?name=${viz.name.toLocaleLowerCase()}` }}>
                details
              </Link>
            </Button>
          </Grid.Row>
          <Grid.Row>
            <Button basic={true}>
              <Link to={{ pathname: '/dataset', search: `?viz=${viz.name.toLocaleLowerCase()}` }}>launch</Link>
            </Button>
          </Grid.Row>
        </Grid.Column>
      </Grid>
    );
  }
}
