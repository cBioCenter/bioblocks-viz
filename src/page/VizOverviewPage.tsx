import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Accordion, Button, Container, Divider, Grid, Header, Icon, List } from 'semantic-ui-react';

import { Link } from 'react-router-dom';
import { IVizExample, IVizOverviewData, VizData } from '~chell-viz~/data';

export interface IVizOverviewPageProps extends Partial<RouteComponentProps> {}

export interface IVizOverviewPageState {
  currentViz: IVizOverviewData | null;
}

export class VizOverviewPage extends React.Component<IVizOverviewPageProps, IVizOverviewPageState> {
  constructor(props: IVizOverviewPageProps) {
    super(props);
    this.state = {
      currentViz: null,
    };
  }

  public componentDidMount() {
    this.setupCurrentViz();
  }

  public componentDidUpdate(prevProps: IVizOverviewPageProps) {
    if (this.props.location !== prevProps.location) {
      this.setupCurrentViz();
    }
  }

  public render() {
    return (
      this.state.currentViz && (
        <Container>
          {this.renderOverview(this.state.currentViz)}
          {this.renderExamples(this.state.currentViz.examples)}
        </Container>
      )
    );
  }

  protected renderOverview(viz: IVizOverviewData) {
    return (
      <Grid centered={true} columns={2}>
        <Grid.Column width={2}>
          <img
            alt={`icon for ${viz.name}`}
            src={`assets/icons/${viz.name.toLocaleLowerCase()}-icon.png`}
            style={{ height: '150px', padding: '20px' }}
          />
        </Grid.Column>
        <Grid.Column textAlign={'left'}>
          <Header as={'h1'}>
            {viz.name}
            <Header.Subheader>{viz.authors.join(', ')}</Header.Subheader>
          </Header>
          <>
            <p>{viz.detailedSummary}</p>
            <List>
              <List.Item>applicable data: {viz.relevantData}</List.Item>
              <List.Item>compatible with: {viz.compatibility.join(', ')}</List.Item>
              <List.Item>
                citation(s):{' '}
                {viz.citations.map((citation, index) => (
                  <React.Fragment key={`${viz.name.toLocaleLowerCase()}-citation-${index}`}>
                    {citation.fullCitation} ({<a href={citation.link}>link</a>})
                  </React.Fragment>
                ))}
              </List.Item>
              <List.Item>
                version: {viz.repo.version} (last updated {viz.repo.lastUpdate}),
                {<a href={viz.repo.link}> github link</a>}
              </List.Item>
              <List.Item>
                <Grid.Column floated={'right'}>
                  <Button basic={true} icon={true} labelPosition={'right'}>
                    <Link to={{ pathname: '/dataset', search: `?viz=${viz.name.toLocaleLowerCase()}` }}>
                      {`launch ${viz.name}`}
                    </Link>
                    {/* Power Gap */}
                    <Icon name={'external alternate'} />
                  </Button>
                </Grid.Column>
              </List.Item>
            </List>
          </>
        </Grid.Column>
      </Grid>
    );
  }

  protected renderExamples(examples: IVizExample[]) {
    const panels = [
      {
        content: {
          content: (
            <>
              <Divider />
              <Grid centered={true} container={true} divided={'vertically'} columns={1}>
                {examples.map((example, index) => (
                  <Grid.Row columns={2} key={`viz-example-${index}`}>
                    {this.renderExampleEntry(example)}
                  </Grid.Row>
                ))}
              </Grid>
            </>
          ),
        },
        key: 'examples',
        title: 'examples',
      },
      {
        content: 'coming soon!',
        key: 'learn',
        title: 'learn',
      },
    ];

    return <Accordion panels={panels} defaultActiveIndex={0} />;
  }

  protected renderExampleEntry(example: IVizExample) {
    return (
      <>
        <Grid.Column width={2}>
          <img src={example.icon} alt={`${example.name} icon`} style={{ height: '75px', width: '75px' }} />
        </Grid.Column>
        <Grid.Column textAlign={'left'} width={8}>
          <Header>{example.name}</Header>
          <p>{example.summary}</p>
        </Grid.Column>
        <Grid.Column floated={'right'}>
          <Button basic={true} icon={true} labelPosition={'right'}>
            <Link to={example.link}>{'launch example'}</Link>
            {/* Power Gap */}
            <Icon name={'external alternate'} />
          </Button>
        </Grid.Column>
      </>
    );
  }

  protected setupCurrentViz() {
    const params = new URLSearchParams(this.props.location ? this.props.location.search : '');
    // tslint:disable-next-line:no-backbone-get-set-outside-model
    const vizName = params.get('name');

    if (vizName === VizData.spring.name.toLocaleLowerCase()) {
      this.setState({
        currentViz: VizData.spring,
      });
    } else if (vizName === VizData.tfjsTsne.name.toLocaleLowerCase()) {
      this.setState({
        currentViz: VizData.tfjsTsne,
      });
    } else if (vizName === VizData.anatomogram.name.toLocaleLowerCase()) {
      this.setState({
        currentViz: VizData.anatomogram,
      });
    }
  }
}
