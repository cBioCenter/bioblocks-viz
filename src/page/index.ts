export * from './AppsPage';
export * from './DatasetPage';
export * from './LandingPage';
export * from './VignettesPage';
export * from './VizOverviewPage';
