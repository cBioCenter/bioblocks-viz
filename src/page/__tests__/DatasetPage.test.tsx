import { shallow } from 'enzyme';
import * as React from 'react';

import { DatasetPage } from '~chell-viz~/page';

describe('DatasetPage', () => {
  it('Should match existing snapshot.', () => {
    const wrapper = shallow(<DatasetPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
