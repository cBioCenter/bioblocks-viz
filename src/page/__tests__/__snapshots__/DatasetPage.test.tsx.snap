// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetPage Should match existing snapshot for initial visualizations. 1`] = `
<div
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <Grid
    centered={true}
    columns={2}
    padded={true}
    stackable={true}
    stretched={false}
  >
    <GridColumn
      key="dataset-visualization-0"
      style={
        Object {
          "width": "auto",
        }
      }
    >
      <Connect(Anatomogram) />
    </GridColumn>
  </Grid>
</div>
`;

exports[`DatasetPage Should match existing snapshot for initial visualizations. 2`] = `
<div
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <Grid
    centered={true}
    columns={2}
    padded={true}
    stackable={true}
    stretched={false}
  >
    <GridColumn
      key="dataset-visualization-0"
      style={
        Object {
          "width": "auto",
        }
      }
    >
      <Connect(Anatomogram) />
    </GridColumn>
    <GridColumn
      key="dataset-visualization-1"
      style={
        Object {
          "width": "auto",
        }
      }
    >
      <Message
        error={true}
      >
        Currently unsupported visualization 'empty'
      </Message>
    </GridColumn>
  </Grid>
</div>
`;

exports[`DatasetPage Should match existing snapshot for initial visualizations. 3`] = `
<div
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <Grid
    centered={true}
    columns={2}
    padded={true}
    stackable={true}
    stretched={false}
  >
    <GridColumn
      key="dataset-visualization-0"
      style={
        Object {
          "width": "auto",
        }
      }
    >
      <Connect(SPRING)
        datasetLocation="{hpc/sample}"
        isFullPage={true}
      />
    </GridColumn>
  </Grid>
</div>
`;

exports[`DatasetPage Should match existing snapshot for initial visualizations. 4`] = `
<div
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <Grid
    centered={true}
    columns={2}
    padded={true}
    stackable={true}
    stretched={false}
  >
    <GridColumn
      key="dataset-visualization-0"
      style={
        Object {
          "width": "auto",
        }
      }
    >
      <Connect(SPRING)
        datasetLocation="{mouse/sample}"
        isFullPage={false}
      />
    </GridColumn>
    <GridColumn
      key="dataset-visualization-1"
      style={
        Object {
          "width": "auto",
        }
      }
    >
      <Message
        error={true}
      >
        Currently unsupported visualization 'empty'
      </Message>
    </GridColumn>
  </Grid>
</div>
`;

exports[`DatasetPage Should match existing snapshot for initial visualizations. 5`] = `
<div
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <Grid
    centered={true}
    columns={2}
    padded={true}
    stackable={true}
    stretched={false}
  >
    <GridColumn
      key="dataset-visualization-0"
      style={
        Object {
          "width": "auto",
        }
      }
    >
      <Connect(tSNE - TensorFlow)
        datasetLocation="{hpc/sample}"
        isFullPage={true}
      />
    </GridColumn>
  </Grid>
</div>
`;

exports[`DatasetPage Should match existing snapshot for initial visualizations. 6`] = `
<div
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <Grid
    centered={true}
    columns={2}
    padded={true}
    stackable={true}
    stretched={false}
  >
    <GridColumn
      key="dataset-visualization-0"
      style={
        Object {
          "width": "auto",
        }
      }
    >
      <Connect(tSNE - TensorFlow)
        datasetLocation="{mouse/sample}"
        isFullPage={false}
      />
    </GridColumn>
    <GridColumn
      key="dataset-visualization-1"
      style={
        Object {
          "width": "auto",
        }
      }
    >
      <Message
        error={true}
      >
        Currently unsupported visualization 'empty'
      </Message>
    </GridColumn>
  </Grid>
</div>
`;

exports[`DatasetPage Should match existing snapshot when changing the visualization. 1`] = `
<div
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <Grid
    centered={true}
    columns={2}
    padded={true}
    stackable={true}
    stretched={false}
  />
</div>
`;

exports[`DatasetPage Should match existing snapshot when no props are provided. 1`] = `
<div
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <Grid
    centered={true}
    columns={2}
    padded={true}
    stackable={true}
    stretched={false}
  />
</div>
`;
